addons/binding/org.openhab.binding.dscalarm/src/main/java/org/openhab/binding/dscalarm/internal/discovery/IT100BridgeDiscovery.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dscalarm.internal.discovery;
+
+import gnu.io.CommPortIdentifier;
+
+import java.util.Enumeration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is responsible for discovering the DSC IT100 RS232 Serial interface.
+ * 
+ * @author Russell Stephens - Initial Contribution
+ * 
+ */
+public class IT100BridgeDiscovery {
+	private final static Logger logger = LoggerFactory.getLogger(IT100BridgeDiscovery.class);
+
+	private DSCAlarmBridgeDiscovery dscAlarmBridgeDiscovery = null;
+
+	/**
+	 * Constructor.
+	 */
+	public IT100BridgeDiscovery(DSCAlarmBridgeDiscovery dscAlarmBridgeDiscovery) {
+		this.dscAlarmBridgeDiscovery = dscAlarmBridgeDiscovery;
+	}
+
+	/**
+	 * Method for Bridge Discovery.
+	 */
+	public synchronized void discoverBridge() {
+		logger.debug("Starting IT-100 Bridge Discovery.");
+
+	    Enumeration<?> ports = CommPortIdentifier.getPortIdentifiers();
+
+	    while (ports.hasMoreElements()) {
+	        CommPortIdentifier port = (CommPortIdentifier) ports.nextElement();
+        	System.out.println("discoverBridge(): Port '" + port.getName() + "' Found!");
+	        if (port.getPortType() == CommPortIdentifier.PORT_SERIAL) {
+	        	System.out.println("discoverBridge(): Serial Port '" + port.getName() + "' Found!");
+	        	logger.debug("discoverBridge(): Serial Port '{}' Found!",port.getName());
+	        	dscAlarmBridgeDiscovery.addIT100Bridge(port.getName());
```

Comment:
you are taking ANY existing serial port here and adding it as a discovery result - this is not ok. You should only create a discovery result in case you have identified the IT-100 on the port.